src/Extension/FluentExtension.php

```
@@ -679,6 +685,86 @@ public function onAfterDuplicate($original, $doWrite, $relations): void
+        }
+
+        $locales = [];
+
+        /** @var RecordLocale $localeInformation */
+        foreach ($owner->Locales() as $localeInformation) {
+            $sourceLocale = $localeInformation->getSourceLocale();
+            $modelLocale = $localeInformation->getLocaleObject();
+
+            if (!$sourceLocale) {
+                // We don't have any source locale, so we can bail out
+                continue;
+            }
+
+            if ($modelLocale->Locale !== $sourceLocale->Locale) {
+                // Source of this locale is different from current locale, so we can skip it
+                // as this locale content is being inherited
+                continue;
+            }
+
+            if ($modelLocale->Locale === $currentLocale) {
+                // Current locale can be skipped as it was already handled correctly
+                continue;
+            }
+
+            // Add locale which uses current locale as a source to our list
+            $locales[] = $modelLocale->Locale;
+        }
+
+        // No locales need to be actioned
+        if (count($locales) === 0) {
+            return;
+        }
+
+        foreach ($locales as $locale) {
+            FluentState::singleton()->withState(
+                static function (FluentState $state) use ($owner, $locale, $copyRelations): void {
+                    $state->setLocale($locale);
+
+                    $localisedOwner = DataObject::get($owner->ClassName)->byID($owner->ID);
+
+                    // Couldn't find localised data to work with
+                    if (!$localisedOwner->exists()) {
+                        return;
+                    }
+
+                    // Duplicate all localised relations
+                    foreach ($copyRelations as $relation) {
+                        $original = $localisedOwner->{$relation}();
+
+                        if (!$original instanceof DataObject) {
+                            continue;
+                        }
```

How would this ever _not_ be true? Are we silently failing for incorrect configuration here? Or are we saying we only care about `has_one` and `belongs_to` - not the other relation types (if so refer to comment about using `$copyHasOneRelations`) - though if `belongs_to` matters then it probably matters for the current locale too so that needs updating? Or something else?

---

Localised copy currently only supports `has_one`. This check is likely not needed, can remove it if needed.

---

I would prefer this be removed if it's not necessary. It makes it look like something more complex is happening or possible here.